src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Give\Form\LegacyConsumer\Commands;
+
+use Give\Framework\FieldsAPI\FieldCollection;
+use Give\Form\LegacyConsumer\FieldView;
+
+/**
+ * Persist custom field values as donation meta.
+ *
+ * @unreleased
+ */
+class SetupFieldPersistance implements HookCommandInterface {
+	public function __invoke( $hook ) {
+		add_action(
+			'give_insert_payment',
+			function( $donationID, $donationData ) use ( $hook ) {
+				$fieldCollection = new FieldCollection( 'root' );
+				do_action( "give_fields_$hook", $fieldCollection, $donationData['give_form_id'] );
+				$fieldCollection->walk(
+					function( $field ) use ( $donationID ) {
+						if ( isset( $_POST[ 'give_' . $field->getName() ] ) ) {
+							$value = wp_strip_all_tags( $_POST[ 'give_' . $field->getName() ], true );
+							give_update_payment_meta( $donationID, '_give_' . $field->getName(), $value );
```

Comment:
I don't think we need to apply the `_give_` prefix to the field meta as it's not actually us that's adding the meta. Other folks are adding these fields, so just using the field name as-is is fine.

Reply:
See 4c97397